In [1]:
# -*- coding: utf8 -*-

In [2]:
import pandas as pd
import itertools as it
import random as rd
import numpy as np
import matplotlib.pyplot as plt


In [4]:
repHDR="/Users/gilles/ownCloud/Recherche/Boye/HDR/Memoire/figs/"


In [6]:
nomRepertoire="/Volumes/gilles/Transfert/Copies-iMac-GB/Python/phonemisation/"
nomLexiqueOrg=nomRepertoire+"Lexique380-UTF8.txt"
nomRepertoire="/Users/gilles/Dropbox/Travail/Recherche/BoKu/LingInfo/Lexique3/Lexique381/"
nomLexiqueOrg=nomRepertoire+"Lexique381.csv"
Lexique3=pd.read_csv(nomLexiqueOrg,sep="\t")


In [ ]:
noms3=Lexique3[Lexique3["4_cgram"]=="NOM"][["1_ortho","3_lemme","5_genre","6_nombre"]]

In [78]:
nbNoms=len(noms3)
nbNoms

48287

- nombre de lemmes de noms

In [118]:
tableauMS=[]
for g,n in it.product("mfx","spx"):
    genre=(noms3["5_genre"]==g)
    nombre=(noms3["6_nombre"]==n)
    if g=="x":
        genre=(noms3["5_genre"].isnull())
    if n=="x":
        nombre=(noms3["6_nombre"].isnull())
    tableauMS.append((g,n, len(noms3[genre&nombre])))
dfTableauMS=pd.DataFrame.from_records(tableauMS)
dfMS=dfTableauMS.pivot(index=0, columns=1, values=2)
del dfMS.index.name
del dfMS.columns.name

In [122]:
dfMS

p      s     x
f  7069  11787    69
m  9673  15352  1719
x  1110   1500     8

In [44]:
noms3Lemmes=list(noms3.groupby(by=["3_lemme","5_genre"]))


30541


In [62]:
nbNomsLemmes=len(noms3Lemmes)
noms3Lemmes[4000]

(('brouille', 'f'),          1_ortho   3_lemme 5_genre 6_nombre
 18129   brouille  brouille       f        s
 18136  brouilles  brouille       f        p)

In [63]:
noms3SansNombre=noms3[noms3["6_nombre"].isnull()]
nbNomsSansNombre=len(noms3SansNombre)
print "noms sans nombre",len(noms3SansNombre)
noms3SansGenre=noms3[noms3["5_genre"].isnull()]
nbNomsSansGenre=len(noms3SansGenre)
print "noms sans genre",len(noms3SansGenre)


noms sans nombre 1796
noms sans genre 2618


In [ ]:
noms3Specifies=noms3[~noms3[["5_genre","6_nombre"]].isnull().any(axis=1)]
noms3Homographes=noms3[noms3[["5_genre","6_nombre"]].isnull().any(axis=1)]

In [64]:
nbNomsSpecifies=len(noms3Specifies)
nbNomsHomographes=len(noms3Homographes)

In [65]:
print u"noms",nbNoms
print u"noms spécifiés",nbNomsSpecifies
print "noms ambigus",nbNomsHomographes
print "%% de noms ambigus",nbNomsHomographes*1./nbNoms*100

noms 48287
noms spécifiés 43881
noms ambigus 4406
%% de noms ambigus 9.12460910804


- 30541 noms (lemme, genre)
    - 48287 formes
    - 43881 formes spécifiées
    - 4406 formes ambiguës : 9,12%
        - 1796 formes sans nombre
        - 2618 formes sans genre
   

In [86]:
noms3Masc=noms3[noms3["5_genre"]=="m"]
nbNomsMascSpec=len(noms3Masc)
nbNomsMasc=nbNomsMascSpec+nbNomsSansGenre
noms3Fem=noms3[noms3["5_genre"]=="f"]
nbNomsFemSpec=len(noms3Fem)
nbNomsFem=nbNomsFemSpec+nbNomsSansGenre

In [79]:
print 1.0*nbNomsSansGenre/nbNomsMasc
print 1.0*nbNomsSansGenre/nbNomsFem
print 1.0*nbNomsMasc/nbNoms

0.0891628635652
0.121524393074
0.608072566115


In [74]:
noms3Sg=noms3[noms3["6_nombre"]=="s"]
nbNomsSgSpec=len(noms3Sg)
nbNomsSg=nbNomsSgSpec+nbNomsSansNombre
noms3Pl=noms3[noms3["6_nombre"]=="p"]
nbNomsPlSpec=len(noms3Pl)
nbNomsPl=nbNomsPlSpec+nbNomsSansNombre


In [91]:
print 1.0*nbNomsSansNombre/nbNomsSg
print 1.0*nbNomsSansNombre/nbNomsPl
print 1.0*nbNomsSgSpec/nbNoms,1.0*nbNomsSg/nbNoms
print 1.0*nbNomsPlSpec/nbNoms,1.0*nbNomsPl/nbNoms

0.0590110070642
0.0914087947883
0.593099592023 0.630293867915
0.369706132085 0.406900407977


In [101]:
noms3MascSansNombre=noms3SansNombre[(noms3SansNombre["5_genre"]=="m")]
nbNomsMascSansNombre=len(noms3MascSansNombre)
noms3FemSansNombre=noms3SansNombre[(noms3SansNombre["5_genre"]=="f")]
nbNomsFemSansNombre=len(noms3FemSansNombre)
print nbNomsMascSansNombre,nbNomsFemSansNombre

1719 69


In [89]:
print 1.0*nbNomsMascSansNombre/nbNomsMasc
print 1.0*nbNomsFemSansNombre/nbNomsFem

0.0585450582385
0.00320289653252


In [97]:
nbNomsMascSansNombreSXZ=len(noms3MascSansNombre[noms3MascSansNombre["1_ortho"].str.contains("[sxz]$")])
print 1.0*nbNomsMascSansNombreSXZ/nbNomsMascSansNombre

0.694589877836


In [103]:
nbNomsSansGenreSgSpec=len(noms3SansGenre[noms3SansGenre["6_nombre"]=="s"])
nbNomsSansGenrePlSpec=len(noms3SansGenre[noms3SansGenre["6_nombre"]=="p"])
print nbNomsSansGenreSgSpec,nbNomsSansGenrePlSpec,nbNomsSansGenre

1500 1110 2618


In [99]:
nbNomsFemSansNombre,nbNomsMascSansNombre,nbNomsSansNombre

(69, 1719, 1796)